ReactiveCocoa/Objective-C/RACSequence.m

```diff
@@ -219,7 +219,11 @@ - (id)foldRightWithStart:(id)start reduce:(id (^)(id, RACSequence *))reduce {
 	if (self.head == nil) return start;
 	
 	RACSequence *rest = [RACSequence sequenceWithHeadBlock:^{
-		return [self.tail foldRightWithStart:start reduce:reduce];
+		if (self.tail) {
+			return [self.tail foldRightWithStart:start reduce:reduce];
+		} else {
+			return start;
+		}
```

Comment:
I'm a bit confused about how `RACSequence` is supposed to work.

The documentation for `-[RACSequence tail]` says this:

```
/// The first object in the sequence, or nil if the sequence is empty.
///
/// Subclasses must provide an implementation of this method.
@property (nonatomic, strong, readonly) id head;
```

That makes me think that the existing code here is correct. If `self.head` is non-nil (we check above), then `self.tail` should also be non-nil. Only if `self.head` is nil should `self.tail` be nil (an empty sequence).

`-[RACUnarySequence tail]` is hard-coded to return `nil`, however. Changing it to return `RACSequence.empty` does make this test pass. However, another test fails:

```
	qck_it(@"should use the values from the head and tail blocks", ^{
		expect(sequence.head).to(equal(@0));
		expect(sequence.tail.head).to(equal(@1));
		expect(sequence.tail.tail).to(beNil());
	});
```

Since I don't think we want to change the behavior at this point, I suppose this fix is correct. However, it'd be nice to correct/clarify the `-[RACSequence tail]` documentation. Would you be up for doing that here?